.github/workflows/integrationTests.yaml

```diff
@@ -10,8 +10,43 @@ on:
 
 
 jobs:
+  check:
+    name: Check IT-disabling files
+    outputs:
+      run_job: ${{ steps.check_files.outputs.run_job }}
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+      - name: check modified files
+        id: check_files
+        run: |
+          git diff --name-only HEAD^ HEAD > files.txt
+          while IFS= read -r file
+          do
+            if [[ $file != generated/* ]] && [[ $file != generator/* ]]; then
+              echo "run_job=true" >> $GITHUB_OUTPUT
+              break
+            else
+              echo "run_job=false" >> $GITHUB_OUTPUT
```

Comment:
You could skip the `else` by setting `run_job=false` before the loop. Then instead of getting written on each "excluded" file, it defaults to false and only gets changed once if a non-excluded file is found.